registry/client/auth/authchallenge.go

```diff
@@ -8,13 +9,23 @@ import (
 // Octet types from RFC 2616.
 type octetType byte
 
-// authorizationChallenge carries information
-// from a WWW-Authenticate response header.
-type authorizationChallenge struct {
-	Scheme     string
+// Challenge carries information from a WWW-Authenticate response header.
+// See RFC 2617.
+type Challenge struct {
+	// Scheme is the auth-scheme according to RFC 2617
+	Scheme string
+
+	// Parameters are the auth-params according to RFC 2617
 	Parameters map[string]string
 }
 
+// APIVersion represents a version of an API including its
+// type and version number.
+type APIVersion struct {
+	Type    string
+	Version string
+}
+
 var octetTypes [256]octetType
```

Comment:
Can we put the parse related stuff into a `parser.go` file? Perhaps a comment here, would help, as well.